packages/discord-request/src/manager.ts

```diff
@@ -113,8 +124,28 @@ export class Manager {
 		this.startSweepers();
 	}
 
-	get globalTimeout() {
-		return this.globalReset + OFFSET - Date.now();
+	async getGlobalReset() {
```

Comment:
Hm, moving these all from `get` to async functions is rough. Totally get why it has to happen, but I kind of want to think about if a different organization / abstraction pattern would clean up some of this. 

It is nice that as far as the user is concerned there is no change, so maybe I'm overly concerned about this.